iChor/iChorCNA_hg19/iChorCNA_hg19.wdl

```diff
@@ -1,327 +1,349 @@
+param_dict = {a: b for a, b in params}
+with open("tumor_fraction", "w") as p:
+    p.write(param_dict["Tumor Fraction"])
+with open("ploidy", "w") as p:
+    p.write(param_dict["Ploidy"])
+with open("subclone_fraction", "w") as p:
+    p.write(param_dict["Subclone Fraction"])
+with open("fraction_genome_subclonal", "w") as p:
+    p.write(param_dict["Fraction Genome Subclonal"])
+with open("fraction_cna_subclonal", "w") as p:
+    p.write(param_dict["Fraction CNA Subclonal"])
+with open("gc-map_correction_mad", "w") as p:
+    p.write(param_dict["GC-Map correction MAD"])
+with open("top_solution_log_likelihood", "w") as p:
+    p.write(str(log_lik))
 
-	runtime {
-		docker: "python:3"
-    	memory: "2 GB"
-    	disks: "local-disk 10 HDD"
-    	preemptible: 1
-    	maxRetries: 1
-	}
+CODE
+    >>>
+
+    runtime {
+        docker: "python:3"
+        memory: "2 GB"
+        disks: "local-disk 10 HDD"
+        preemptible: 1
+        maxRetries: 1
+    }
 
     output {
-    	Float tumor_fraction = read_float("tumor_fraction")
-    	Float ploidy = read_float("ploidy")
-    	String subclone_fraction = read_string("subclone_fraction")
-    	String fraction_genome_subclonal = read_string("fraction_genome_subclonal")
-    	String fraction_cna_subclonal = read_string("fraction_cna_subclonal")
-    	String gc_map_correction_mad = read_string("gc-map_correction_mad")
+        Float tumor_fraction = read_float("tumor_fraction")
+        Float ploidy = read_float("ploidy")
+        String subclone_fraction = read_string("subclone_fraction")
+        String fraction_genome_subclonal = read_string("fraction_genome_subclonal")
+        String fraction_cna_subclonal = read_string("fraction_cna_subclonal")
+        String gc_map_correction_mad = read_string("gc-map_correction_mad")
         Int top_solution_log_likelihood = read_int("top_solution_log_likelihood")
     }
 }
 
 task bundlePerChromosomePlots {
-	input {
-	   Array[File] chrom_plots
-	   String sample_id
+    input {
+        Array[File] chrom_plots
+        String sample_id
 
-	   Int memGB = 2
-	   Int diskGB = 10
-	   Int preemptible = 3
-	   Int maxRetries = 1
+        Int memGB = 2
+        Int diskGB = 10
+        Int preemptible = 3
+        Int maxRetries = 1
 
-	   String? docker_override
-	}
+        String docker_override = "us.gcr.io/tag-team-160914/tag-tools:1.0.0"
+    }
 
-   command <<<
-      set -e
-      CHROM_PLOTS=`ls ${sep=" " chrom_plots} | sort -V`
+    command <<<
+        set -euxo pipefail
+        
+        # load list of plots into an array, which will correctly handle files with spaces in them
+        readarray -t CHROM_PLOTS < <(sort -V "~{write_lines(chrom_plots)}")
 
-      gs -dBATCH -dNOPAUSE -q -sDEVICE=pdfwrite \
-         -sOutputFile="${sample_id}_OptimalSolutionPerChrom.pdf" \
-         $CHROM_PLOTS
+        gs -dBATCH -dNOPAUSE -q -sDEVICE=pdfwrite \
+            -sOutputFile="~{sample_id}_OptimalSolutionPerChrom.pdf" \
+            "${CHROM_PLOTS[@]}"
```

Comment:
This section is similar to the previous version, but
1. `write_lines` is used instead of `ls` which can have weird behavior in some edge cases.
2. The values are read into an array which will prevent word-splitting (e.g. in case someone runs a crazy sample ID).